src/main/java/org/broadinstitute/hellbender/tools/funcotator/engine/FuncotationEngine.java

```diff
@@ -0,0 +1,108 @@
+package org.broadinstitute.hellbender.tools.funcotator.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.tools.funcotator.*;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.DataSourceUtils;
+import org.broadinstitute.hellbender.tools.funcotator.dataSources.gencode.GencodeFuncotation;
+import org.broadinstitute.hellbender.tools.funcotator.metadata.FuncotationMetadata;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class FuncotationEngine {
+
+    private static final Logger logger = LogManager.getLogger(FuncotationEngine.class);
+
+    private FuncotationMetadata inputMetadata;
+    private List<DataSourceFuncotationFactory> dataSourceFactories;
+
+    public FuncotationEngine(final FuncotationMetadata metadata, final List<DataSourceFuncotationFactory> funcotationFactories) {
```

Comment:
Add javadoc for all public methods in this class.

Reply:
Fixed!